smart_contracts/contracts/explorer/faucet-stored/src/main.rs

```diff
@@ -3,57 +3,245 @@
+    // the faucet installer does not want the ENTRY_POINT_FAUCET to be called directly by users for
+    // security reasons. Another case would be where this contract is deployed to a private Casper
+    // Network where all users are trusted to use the faucet to issue themselves token
+    // distributions responsibly.
+    named_keys.insert(
+        faucet::AUTHORIZED_ACCOUNT.to_string(),
+        storage::new_uref(None::<PublicKey>).into(),
+    );
+
+    (named_keys, purse)
+}
+
+#[no_mangle]
+pub extern "C" fn call() {
+    let id: u64 = runtime::get_named_arg(faucet::ARG_ID);
+
     let entry_points = {
         let mut entry_points = EntryPoints::new();
 
-        let entry_point = EntryPoint::new(
-            ENTRY_POINT_NAME,
+        let faucet = EntryPoint::new(
+            faucet::ENTRY_POINT_FAUCET,
             vec![
-                Parameter::new(ARG_TARGET, AccountHash::cl_type()),
-                Parameter::new(ARG_AMOUNT, CLType::U512),
+                Parameter::new(faucet::ARG_ID, CLType::Option(Box::new(CLType::U64))),
+                Parameter::new(faucet::ARG_TARGET, CLType::PublicKey),
+                Parameter::new(faucet::ARG_AMOUNT, CLType::U512),
             ],
             CLType::Unit,
             EntryPointAccess::Public,
-            EntryPointType::Session,
+            EntryPointType::Contract,
         );
 
-        entry_points.add_entry_point(entry_point);
+        let set_variables = EntryPoint::new(
+            faucet::ENTRY_POINT_SET_VARIABLES,
+            vec![
+                Parameter::new(
+                    faucet::ARG_AVAILABLE_AMOUNT,
+                    CLType::Option(Box::new(CLType::U512)),
+                ),
+                Parameter::new(
+                    faucet::ARG_TIME_INTERVAL,
+                    CLType::Option(Box::new(CLType::U64)),
+                ),
+                Parameter::new(
+                    faucet::ARG_DISTRIBUTIONS_PER_INTERVAL,
+                    CLType::Option(Box::new(CLType::U64)),
+                ),
+            ],
+            CLType::Unit,
+            EntryPointAccess::Public,
+            EntryPointType::Contract,
+        );
+
+        let authorize_to = EntryPoint::new(
+            faucet::ENTRY_POINT_AUTHORIZE_TO,
+            vec![Parameter::new(
+                faucet::ARG_TARGET,
+                CLType::Option(Box::new(CLType::PublicKey)),
+            )],
+            CLType::Unit,
+            EntryPointAccess::Public,
+            EntryPointType::Contract,
+        );
+
+        entry_points.add_entry_point(faucet);
+        entry_points.add_entry_point(set_variables);
+        entry_points.add_entry_point(authorize_to);
 
         entry_points
     };
-    storage::new_contract(
+
+    // The installer will create the faucet purse and give it to the newly installed
+    // contract so that the installing account and the newly installed account will
```

Comment:
"installing account and the newly installed account" do we mean, the newly installed contract here?

Reply:
Done in [98e09fd](https://github.com/casper-network/casper-node/pull/2976/commits/98e09fd19990fba30486058426258f4fafe72349).